release_creation/snapshot/create.py

```diff
@@ -0,0 +1,93 @@
+from typing import Dict, List, Optional, Tuple
+from semantic_version import Version
+import os, platform
+import platform
+import subprocess
+import shutil
+
+_OUTPUT_ARCHIVE_FILE_BASE = "dbt-core-all-adapters-snapshot"
+_FILE_DIR = os.path.dirname(os.path.realpath(__file__))
+
+
+def _get_local_os() -> str:
+    local_sys = platform.system()
+    if local_sys == "Linux":
+        return "linux"
+    elif local_sys == "Windows":
+        return "windows"
+    elif local_sys == "Darwin":
+         return "mac"
+    else:
+        raise ValueError(f"Unsupported system {local_sys}")
+
+def _get_extra_platforms_for_os(_os: str) -> List[str]:
+    if _os == "mac":
+        return ['macosx_10_9_x86_64', 'macosx_11_0_arm64']
+    else:
+        return ['manylinux_2_17_x86_64','manylinux2014_x86_64']
+
+
+def _get_requirements_prefix(
+    major_version: Optional[int], minor_version: Optional[int], is_pre: bool = False
+):
+    suffix = "latest"
+    if is_pre:
+        suffix = "pre"
+    return f"v{major_version}.{minor_version}.{suffix}"
+
+
+def _generate_download_command_args(
+    requirements_prefix: str, is_pre: bool = False
+) -> str:
+    download_args = []
+    if is_pre:
+        download_args.append(" --pre")
+    download_args.append(
+        f" -r {_FILE_DIR}/requirements/{requirements_prefix}.requirements.txt")
+    return "".join(download_args)
+
+
+def generate_snapshot(target_version: Version) -> Dict[str, str]:
+    """creates a zip archive of the python dependencies for the provided 
+       semantic version
+
+    Args:
+        target_version (Version): the input version to use when determining 
+        the requirements to download.
+
+    Returns:
+        Dict[str, str]: dict of generated snapshot assets, key is it's name 
+                        and the value is the path to the file.
+    """
+    is_pre = True if target_version.prerelease else False
+    requirements_prefix = _get_requirements_prefix(
+        major_version=target_version.major, minor_version=target_version.minor, is_pre=is_pre
+    )
+    archive_path = f"{_OUTPUT_ARCHIVE_FILE_BASE}-{target_version}"
+    _os = _get_local_os()
```

Comment:
is the '_' prefix here indicating a convention around private/public variables, or avoiding clashing with the `os` module? If it's the latter - `local_os` would be less ambiguous.